### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by benefacto**
Refactoring application into separate classes and files.

### CODE REVIEW

File: src/agentAvailabilityBot.ts [hunk 1]
```diff
@@ -39,27 +39,30 @@ export class AgentAvailabilityBot extends Bot {
...
     }
 
     startUp() {
-        console.log('Starting up', Bot.prototype.myInfo.call(this)?.username, Bot.prototype.myInfo.call(this)?.devicename);
-        console.log(`Watching for new messages to ${Bot.prototype.myInfo.call(this)?.username} starting with ${this.commandPrefix}`);
+        console.log('Starting up', this.myInfo.call(this)?.username, this.myInfo.call(this)?.devicename);
+        console.log(`Watching for new messages to ${this.myInfo.call(this)?.username} starting with ${this.commandPrefix}`);
         const onError = (e: any) => console.error(e);
-        Bot.prototype.chat.watchAllChannelsForNewMessages.apply(this, [this.onMessage, onError]);
+        this.chat.watchAllChannelsForNewMessages.apply(this, [this.onMessage, onError]);
```
Comment (benefacto) [inline src/agentAvailabilityBot.ts]: Need to fix this call; "this" is undefined in the Chat object

**Comment by dOrgJelli** [comment]
Some Random Feedback & Thoughts:
- For consistency sake, I'm going to move the following files to the root directory:
  - package.json
  - tasks.json
  - tsconfig.json
  - yarn.lock
- It's typical to have build/run/test/etc scripts stored in the "scripts" section of your package.json.
- I'm going to add a build command, so users can run `yarn build`
- tasks.json is vscode specific, so I'm going to move that into the .vscode directory
- The version of node & npm each developer uses can drastically change how things are installed and run locally. Because of this, it's important to have the version this project uses written somewhere. Two options exist: package.json's engines field, or an .nvmrc file. I like the latter, so I added an .nvmrc file to the project's root that just simply states `v10.16.3`. This is the version of node. If you install `nvm` on your machine (node version manager), you can run `nvm install` and `nvm use` in any folder that has an .nvmrc file, and you're ready to run `yarn` and get expected behavior.

**Comment by dOrgJelli** [comment]
Some Random Feedback & Thoughts:
- In src/index.ts, when executing main(), you're taking the return result `text` and logging it to the console. main does not return text, so this will be undefined. I'm going to remove and improve this main() + error handling logic.
- Instead of assuming src/index.ts will always be used as a CLI, we can also make it work as a module. This will allow other node applications to import this bot as just another module, and use it as they choose. We can do this by executing the main function if `if (require.main === module)`, meaning the "main module" is "this module". Else, we'll just export the AgentAvailabilityBot for others to use.
- `class Availability` is being intialized in `AgentAvailabilityBot` as an interface (without the `new` keyword). This means that if you added methods to the `Availability` class, they wouldn't be callable because the methods are never bound to the new instance. I've made `Availability` an interface for now. It can be made a class in the future if it has methods we want to add to it.

**Comment by dOrgJelli** [comment]
Some Random Feedback & Thoughts:
* In _msgReply, the case where `args.length === 0` was not being handled. I added an error message for this case when the user doesn't give a command.
* It looks like there isn't a help command created yet. I created this issue for this: https://github.com/dOrgTech/AgentAvailability/issues/4
* Having steps to run the project, without using vscode, would be ideal. Additionally having more information about how to test with keybase would be great.
* When there are env variables that are required for a project, I've seen most projects include a `.env.example` file at the root directory of the project. You can also have a `.env.test` that can be used for automated testing (without changes). Currently the env variables are only listed in the README.

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**